llvm/test/CodeGen/AArch64/GlobalISel/arm64-atomic.ll

```diff
@@ -2655,13 +2655,15 @@ define i8 @atomicrmw_max_i8(ptr %ptr, i8 %rhs) {
 define i8 @atomicrmw_umin_i8(ptr %ptr, i8 %rhs) {
 ; CHECK-NOLSE-O1-LABEL: atomicrmw_umin_i8:
 ; CHECK-NOLSE-O1:       ; %bb.0:
-; CHECK-NOLSE-O1-NEXT:    and w9, w1, #0xff
+; CHECK-NOLSE-O1-NEXT:    ; kill: def $w1 killed $w1 def $x1
+; CHECK-NOLSE-O1-NEXT:    and x9, x1, #0xff
 ; CHECK-NOLSE-O1-NEXT:  LBB35_1: ; %atomicrmw.start
 ; CHECK-NOLSE-O1-NEXT:    ; =>This Inner Loop Header: Depth=1
-; CHECK-NOLSE-O1-NEXT:    ldaxrb w8, [x0]
-; CHECK-NOLSE-O1-NEXT:    and w8, w8, #0xff
-; CHECK-NOLSE-O1-NEXT:    cmp w8, w9
-; CHECK-NOLSE-O1-NEXT:    csel w10, w8, w9, lo
+; CHECK-NOLSE-O1-NEXT:    ldaxrb w10, [x0]
```

Comment:
Regression?

Reply:
no LSE!

Reply:
There are several spurious `and` s in file. They are limited to NOLSE-O1 and OUTLINE-O1. O0 and LSE are not affected.

Reply:
`csel w10, w8, w9, lo` changed to`csel x10, x10, x9, ls`  and the `cmp` learned an `uxtb`.

We change predicate in canonicalization of constants to the RHS and in the matchZextOfICmp combine. It was probably the latter combine.

Reply:
The `csel` changed from w to x registers. It was mostly the zext-of-icmp because it changes the width of the underlying G_ICMP.